tests/dummy/app/templates/demo/select.hbs

```diff
@@ -99,4 +98,40 @@
 </p>
 {{code-snippet name="select.async-search.hbs"}}
 
+<h2>Select Header</h2>
+
+<div layout="column" layout-align="center center">
+  <div layout="column" layout-align="center center" style="height: 130px;">
+    {{! BEGIN-SNIPPET select.select-header }}
+  {{#paper-select
+    placeholder="Vegetable"
+    value=selectedVegetable
+    itemLabelCallback=vegetableLabelCallback
+    onChange=(action (mut selectedVegetable))
+    as |items|
+  }}
+    {{#paper-select-header class="demo-select-header"}}
+      {{input value=searchTerm
```

Comment:
We should use 
```hbs
<input
  type="search"
  value={{searchTerm}}
  class="demo-header-searchbox"
  placeholder="Search for a vegetable.."
  oninput={{action "searchKeyPressed" value="target.value"}}>
```

1. `{{input` helpers are being less used because they are two way bound by default
2. we should use `input` instead of `keydown` events because one may input text without typing a key (e.g pasting)